### PR TITLE
164 add obstacles

### PR DESCRIPTION
**Pull request by sir-marc**
Added some maps:

FeastOrFamine:
image

WarpedArena:
image

CrissCross:
image

UnderConstruction:
image

### CODE REVIEW

**Comment by sir-marc** [comment]
A sample script you can run to move around the walls.
Not the finest code, but does the job

```.js
const target = nearestFood

const canMoveTo = (targetPosition) => {
    return !walls.find(wall => wall.position.x === targetPosition.x && wall.position.y === targetPosition.y)
}

const canMoveUp = () => {
    const targetPosition = {
        x: userUnit.position.x,
        y: userUnit.position.y - 1
    }

    return canMoveTo(targetPosition)
}

const canMoveDown = () => {
    const targetPosition = {
        x: userUnit.position.x,
        y: userUnit.position.y + 1
    }

    return canMoveTo(targetPosition)
}

const canMoveRight = () => {
    const targetPosition = {
        x: userUnit.position.x + 1,
        y: userUnit.position.y
    }

    return canMoveTo(targetPosition)
}

const canMoveLeft = () => {
    const targetPosition = {
        x: userUnit.position.x - 1,
        y: userUnit.position.y
    }

    return canMoveTo(targetPosition)
}

const moveCheckMap = {
    'RIGHT': canMoveRight,
    'LEFT': canMoveLeft,
    'UP': canMoveUp,
    'DOWN': canMoveDown,
}
const dodgeMap = {
    'RIGHT': 'UP',
    'LEFT': 'UP',
    'UP': 'RIGHT',
    'DOWN': 'RIGHT',
}
const oppositeMap = {
    'RIGHT': 'LEFT',
    'LEFT': 'RIGHT',
    'UP': 'DOWN',
    'DOWN': 'UP',
}

const tryToMove = direction => {
    if (direction === oppositeMap[store.dodgedTo]) {
        return
    }
    store.dodgedTo = undefined
    if (moveCheckMap[direction]()) {
        move(direction);
        store.couldNotMove = undefined;
    } else {
        store.couldNotMove = direction;
    }
}


if (store.couldNotMove && moveCheckMap[store.couldNotMove]()) {
    store.couldNotMove = undefined;
} 

if (store.couldNotMove) {
    move(dodgeMap[store.couldNotMove])
    store.dodgedTo = dodgeMap[store.couldNotMove];
} else {
    if(target.position.y > userUnit.position.y) {
        tryToMove('DOWN')
    } else if (target.position.y < userUnit.position.y) {
        tryToMove('UP')
    }
    if(target.position.x > userUnit.position.x) {
        tryToMove('RIGHT')
    } else if (target.position.x < userUnit.position.x) {
        tryToMove('LEFT')
    }
}

if (target.position.x === userUnit.position.x && target.position.y === userUnit.position.y) {
    consume()
    store.couldNotMove = false;
}
```

**Comment by mailaenderli** [comment]
Looks good to me (functional) - but developping a script has become much more demanding

File: game-backend/src/map/maps/warpedArena.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { Position } from '@darwin/types';
+
+const warpedArena: Position[] = [
+  // left
+  { x: 3, y: 2 },
```
Comment (StuckiSimon) [inline game-backend/src/map/maps/warpedArena.ts]: General Note: What about a dedicated string format for these?

```
.......oo.......
.......oo.......
```

The import won't be much more difficult but the readability would be improved.
Sorry, to drop your line count though 😅

Comment (michaelschufi) [reply game-backend/src/map/maps/warpedArena.ts]: I second that ^^.

**Comment by michaelschufi** [comment]
Looks good! Thanks for the parser.

**Review by StuckiSimon — APPROVED** [review]
Looks solid 🧱

**Pull request merged**